### ISSUE TITLE
Singularity image not working - NumPy version issue

### ISSUE THREAD

**Issue by sjfleck** [VIB-PSB/ksrates#62]
Hello,
I've used your program plenty of times in the past, but my university is using a new computing cluster system and I need to make a new installation. Unfortunately, I can't install it like before, so I decided to use the singularity image, which I've had some experience with in the past with running on our cluster. It seems like it conatins the wrong version of nmpy

when I run this command:
singularity exec docker://vibpsb/ksrates ksrates init config_Rt.txt

I get this error:
/usr/local/lib/python3.8/dist-packages/scipy/__init__.py:146: UserWarning: A NumPy version >=1.16.5 and <1.23.0 is required for this version of SciPy (detected version 1.24.4
  warnings.warn(f"A NumPy version >={np_minversion} and <{np_maxversion}"
  
The command seemingly finishes, but I get more errors when I run:
singularity exec docker://vibpsb/ksrates ksrates paralogs-ks config_Rt.txt --n-threads 16

I thought the singularity image should have all the dependencies. Any help with this would be greatly appreciated. Thank you

**Comment by Cecilia-Sensalari** [comment]
Hi sjfleck!

Thanks for opening this issue. I just pulled the `latest` singularity container to check the versioning, it contains:
- Python `3.8.10`
- SciPy `1.7.1`
- Numpy `1.21.2`

1) Can you see which software versions you have?
```
singularity shell docker://vibpsb/ksrates
```
```
Singularity> python3
Python 3.8.10 (default, Mar 13 2023, 10:26:41) 
Type "help", "copyright", "credits" or "license" for more information.
>>> import scipy
>>> scipy.__version__
'1.7.1'
>>> import numpy
>>> numpy.__version__
'1.21.2'
```

2) Can you try to execute the `init` command on the `test` dataset, which gave me no errors?
```
cd ksrates/test
singularity exec docker://vibpsb/ksrates ksrates init config_elaeis.txt
```

I also ran the `init` command using the vibpsb/ksrates Docker container on a different computer, completing without errors.

Cheers,
Cecilia

**Comment by sjfleck** [comment]
Thank you for the quick response. I can only use singularity on the compute nodes of my university's cluster. Unfortunately, I can't run commands in real time, making opening an interactive shell difficult. I can only submit jobs containing commands for it to run for me. 

Can you suggest a command to run that will tell me the python, scipy, numpy versions?
I tried this but it didnt' work:
singularity shell docker://vibpsb/ksrates
python3

With my own data and the test data, the generate-config and init commands work (even though I do get the error mentioned above while running the init command). This is the output:

working directory = /vscratch/grp-vaalbert/Fleck_data/Rauvolfia_data/01_ksrates_default
INFO	- - - - - - - - - - - - - - - - - - - - - - - - 
INFO	Setting up the analysis from configuration file
INFO	Mon Oct 28 19:41:46 2024
INFO	- - - - - - - - - - - - - - - - - - - - - - - - 
INFO	Loading parameters and input files
INFO	Checking structural integrity of input Newick tree...
INFO	Checking if sequence data files exist and if sequence IDs are compatible with wgd pipeline...
INFO	Completed
INFO	
INFO	Extracting ortholog trios for rate-adjustment [ortholog_trios_elaeis.tsv]
INFO	- Each divergent species pair will be rate-adjusted with at maximum 4 trios by using the closest outspecies
INFO	  (as required in configuration file field 'maximum_number_outspecies')
INFO	- Total number of trios: 1
INFO	
INFO	Extracting species pairs for wgd ortholog pipeline [ortholog_pairs_elaeis.tsv]
INFO	- Only species pairs required by the selected trios will be considered.
INFO	Ortholog databases empty or not found at the paths provided in the configuration file.
INFO	All of the species pairs will be used to compute their ortholog distribution and its peak.
INFO	- Total number of species pairs not in database(s): 3
INFO	
INFO	Plotting input phylogenetic tree [tree_elaeis.pdf]
INFO	
INFO	Generating the list of wgd paralog and ortholog runs to be manually executed [wgd_runs_elaeis.txt]
INFO	
INFO	All done
All Done!

When I run the paralogs-ks command, I do not get an output. Just a bunch of errors:

**Comment by Cecilia-Sensalari** [comment]
Hi sjfleck,

Sorry to make you wait!

> Unfortunately, I can't run commands in real time, making opening an interactive shell difficult. I can only submit jobs containing commands for it to run for me. Can you suggest a command to run that will tell me the python, scipy, numpy versions?

You could try the following one-liner, which executes in the container some python commands to print the package versions:
```
singularity exec docker://vibpsb/ksrates python3 -c "import sys, numpy, scipy; print(f'Python: {sys.version.split()[0]}, NumPy: {numpy.__version__}, SciPy: {scipy.__version__}')"
```
This is my output:
```
INFO:    Using cached SIF image
Python: 3.8.10, NumPy: 1.21.2, SciPy: 1.7.1
```

Cheers,
Cecilia

**Comment by sjfleck** [comment]
Cecilia,
I was able to solve this issue earlier this morning and I'm still a little confused as to why it messed things up in the first place. In August, I installed something into my user directory by mistake and never deleted it. It was in ~/.local and there was a python3.8 directory that contained NumPy 1.24.4. I deleted that and ksrates started working. I don't exactly understand why it would prioritize that installation over the singularity image, but I guess it's possible. We can close this issue
